docs/content/guides/6.multistore/3.patterns/3.data/1.different-config-same-integration.md

```diff
@@ -6,3 +6,145 @@ navigation:
+      catalogId: 'b2cProductsCatalog', // store specific value
+      // ...
+    },
+  }
+}
+```
+
+```ts
+// B2B store
+// /apps/stores/b2b/storefront-middleware/integrations/sapcc/config.ts
+
+export const config = {
+  configuration: {
+    api: {
+      baseSiteId: 'b2b-site', // store specific value
+      catalogId: 'b2bProductsCatalog', // store specific value
+      // ...
+    },
+  }
+}
+```
+
+After this operation, the file structure will look like this:
+```bash
+apps/
+├── storefront-middleware
+|  ├── integrations
+|  |  └── sapcc
+|  |     └── config.ts # base config will be overridden by stores config
+└── stores
+   ├── b2c
+   |  ├── storefront-middleware
+   |  |  ├── integrations
+   |  |  |  └── sapcc
+   |  |  |     └── config.ts # new file for b2c store 
+   └── b2b
+      └── storefront-middleware
+         └── integrations
+            └── sapcc
+               └── config.ts # new file for b2b store 
+
+```
+
+## Best Practices
+
+### 1. Use Environment Variables
+Instead of creating separate configuration files for each store, leverage **environment variables** where it is possible. Different environment settings can dynamically configure the middleware or storefront application for each store.
+
+This will help avoiding duplicated files and makes your solution more modular, clean, and easy to maintain.
+
+Let's change the previous configuration example to use environmental variables:
+
+```ts
+// base source code, not nested in stores
+// /apps/storefront-middleware/integrations/sapcc/config.ts
+
+const { BASE_SITE_ID, PRODUCT_CATALOG_ID } = process.env;
+
+export const config = {
+  configuration: {
+    api: {
+      baseSiteId: BASE_SITE_ID, // dynamic value
+      catalogId: PRODUCT_CATALOG_ID, // dynamic value
+      // ...
+    },
+  }
+}
+```
+
+For local development, use `.env` files specific to each store. For example:
+
+```plaintext
+# b2c store
+# /apps/stores/b2c/storefront-middleware/.env
+BASE_SITE_ID="b2c-site"
+PRODUCT_CATALOG_ID="b2cProductsCatalog"
+
+# b2b store
+# /apps/stores/b2c/storefront-middleware/.env
+BASE_SITE_ID="b2b-site"
+PRODUCT_CATALOG_ID="b2bProductsCatalog"
+```
+Your application can then load these files accordingly during development.
+
+:::warning
+Ensure that `.env` files are versioned appropriately. To avoid accidentally committing sensitive configurations, use `.env.example` to provide a template and add `.env` files to `.gitignore`.
+:::
+
+### 2. Use small files for overrides
+Where environment variables aren't suitable (like in cases requiring a lot of customization), follow these steps:
+- Separate the **overridable parts** of your configurations into distinct files.
+- Use **file-based inheritance** to share the majority of the configuration and override only what needs to change for individual stores.
+
+This approach ensures your codebase remains clean and maintainable without introducing unnecessary code duplication.
```

Comment:
It would be good to provide here an example. For instance, user could split the eCommerce config into smaller config files (`apiConfig.ts`, etc.) and use the `config.ts` file to compose those configs. We should promote such way of problem solving which might not be trivial for the customers.

Reply:
Yeah, good idea, i'll add it

Reply:
It's added now.